### PR TITLE
ContainmentFieldComponent's garbage destruction can now be disabled

### PR DESCRIPTION
**Pull request by mhamsterr**
## About the PR

ContainmentField component now has a field (destroyGarbage) which controls whenever containment field should vaporize incoming space garbage (SpaceGarbage). Default value is true => all trash still gets destroyed until told not to.
## Why / Balance

Its a nice feature that containment field can destroy small trash that collides with it, but i'd like to have control over when it should and shouldn't evaporate something
## Technical details

`destroyGarbage` in ContainmentField component, `DestroyGarbage` bool property in ContainmentFieldComponent class.
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

Default behaviour:

https://github.com/space-wizards/space-station-14/assets/81412348/498cc146-2bf8-4359-a78c-e7064181b157

"Do not evaporate" behaviour:

https://github.com/space-wizards/space-station-14/assets/81412348/3299b36e-8353-41dc-8c07-b2a2dad9b730

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Shared/Singularity/Components/ContainmentFieldComponent.cs [hunk 1]
```diff
@@ -18,4 +18,10 @@ public sealed partial class ContainmentFieldComponent : Component
     /// </summary>
     [DataField("maxMass")]
     public float MaxMass = 10000f;
+
+    /// <summary>
+    /// Should field vaporize garbage that collides with it?
+    /// </summary>
+    [DataField("destroyGarbage")]
```
Comment (Tayrtahn) [inline Content.Shared/Singularity/Components/ContainmentFieldComponent.cs:25]:
```suggestion
    [DataField]
```
DataField tag can be generated automatically.

**Review by Tayrtahn — CHANGES_REQUESTED** [review]

**Review by Tayrtahn — APPROVED** [review]

**Comment by EmoGarbage404** [comment]
Is this PR adding parity with something? it seems weird that it just deletes stuff when it already has the funnier behavior of just launching things backwards.

**Comment by mhamsterr** [comment]
> Is this PR adding parity with something?

I made this PR just because it seemed weird to me that component that adds ability to contain singulo also transforms entities into portable recyclers (and that you cannot disable this behaviour without making changes in code)

**Comment by EmoGarbage404** [comment]
oh wait i didn't realize this functionality already existed. my bad.

**Pull request merged**